src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double upperBase;
+    private double lowerBase;
+
+    public IsoscelesTrapezoid(double upperBase,double lowerBase, Color color) {
+        super(color);
+        this.upperBase = upperBase;
+        this.lowerBase = lowerBase;
+    }
+
+    @Override
+    public double calculateArea() {
+        return Math.round((upperBase + lowerBase) / 2 * calculateHeight());
```

Comment:
The `calculateArea` method rounds the area, which might not be necessary. Consider returning the precise value of the area without rounding, as rounding can lead to loss of precision.